ipatests/test_integration/test_installation.py

```diff
@@ -1108,6 +1115,79 @@ def run_request(url, expected_stdout=None, expected_stderr=None):
                 expected_stdout=f'href="https://{self.master.hostname}/'
             )
 
+    @master_unistalled
+    def test_hostname_parameter(self):
+        """
+        Test for #2692 ipa-server-install ignores --hostname
+
+        https://pagure.io/freeipa/issue/2692
+        """
+        original_hostname = self.master.hostname
+        new_hostname = "new." + original_hostname
+        ip_address = self.master.ip
+        hosts = self.master.get_file_contents('/etc/hosts')
```

Comment:
Can you please comment why do we need to edit `/etc/hosts` file?

Anyway you should backup it and restore after the test.

Reply:
Sorry, did not notice it is restored in the end.

Reply:
I need to add new hostname to `/etc/hosts`, because it's being looked for there by the installer. If it's not found, installation fails with
```
Unable to resolve host name, check /etc/hosts or DNS name resolution

```


`/etc/hosts/` is being backed-up and restored in `finally` branch.

Reply:
Thank you for explanation.  
If you will be adding more changes, please also add this to comments in the code as this preparation step is not obvious.